packages/next/src/server/next-server.ts

```diff
@@ -1512,36 +1555,63 @@ export default class NextNodeServer extends BaseServer<
       middleware: true,
     })
 
+    const method = (params.request.method || 'GET').toUpperCase()
+    const requestData = {
+      headers: params.request.headers,
+      method,
+      nextConfig: {
+        basePath: this.nextConfig.basePath,
+        i18n: this.nextConfig.i18n,
+        trailingSlash: this.nextConfig.trailingSlash,
+        experimental: this.nextConfig.experimental,
+      },
+      url: url,
+      page,
+      body:
+        method !== 'GET' && method !== 'HEAD'
+          ? (getRequestMeta(params.request, 'clonableBody') as any)
+          : undefined,
+
+      signal: signalFromNodeResponse(params.response.originalResponse),
+      waitUntil: this.getWaitUntil(),
+    }
+    let result:
+      | UnwrapPromise<ReturnType<typeof import('./web/sandbox').run>>
+      | undefined
+
+    // if no middleware info check for Node.js middleware
+    // this is not in the middleware-manifest as that historically
+    // has only included edge-functions, we need to do a breaking
+    // version bump for that manifest to write this info there if
+    // we decide we want to
     if (!middlewareInfo) {
-      throw new MiddlewareNotFoundError()
-    }
+      let middlewareModule
+      middlewareModule = this.loadNodeMiddleware()
```

Comment:
For some reason this returns a promise in my setup. Causing the error `adapterFn` is not a function. Awaiting the `this.loadNodeMiddleware()` solves it for me.

I'm using node 22, in case that matters.